File: Informatică și tehnologia informației/proba practica/rezolvare_2026/Subiectul_I/b.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

// Subprogramul afla_minim primeste prin n numarul de elemente ale unui vector,
// iar prin v elementele vectorului. Returneaza valoarea minima din vector.
long long afla_minim(int n, const long long v[]) {
    long long val_min = v[0];
    for (int i = 1; i < n; ++i) {
        if (v[i] < val_min) {
            val_min = v[i];
        }
    }
    return val_min;
}

int main() {
    ifstream fin("minim.in");
    ofstream fout("minim.out");
    
    if (!fin) {
        cerr << "Eroare la deschiderea fisierului minim.in" << endl;
        return 1;
    }
    
    int n;
    if (!(fin >> n)) return 0;
    
    long long v[100];
    for (int i = 0; i < n; ++i) {
        fin >> v[i];
    }
    
    // Se cere determinarea minimului din prima jumatate a celor n numere.
    // Daca n este impar (ex: 5), prima jumatate poate fi considerata n/2 (2 elemente)
    // sau (n+1)/2 (3 elemente). De regula, in algoritmi se utilizeaza n/2 (impartirea intreaga).
    // Daca n = 1, prima jumatate va avea cel putin 1 element.
    int jumatate = n / 2;
    if (jumatate == 0) jumatate = 1;
    
    long long minim_jumatate = afla_minim(jumatate, v);
    fout << minim_jumatate << "\n";
    
    return 0;
}

File: Informatică și tehnologia informației/proba practica/rezolvare_2026/Subiectul_I/b.pas
program MinimB;

type
  TTablou = array[1..100] of int64;

var
  fin, fout: text;
  n, i, jumatate: integer;
  v: TTablou;

// Subprogramul afla_minim primeste prin n numarul de elemente ale unui vector,
// iar prin v elementele vectorului. Returneaza valoarea minima din vector.
function afla_minim(n: integer; var v: TTablou): int64;
var
  i: integer;
  val_min: int64;
begin
  val_min := v[1];
  for i := 2 to n do
  begin
    if v[i] < val_min then
      val_min := v[i];
  end;
  afla_minim := val_min;
end;

begin
  assign(fin, 'minim.in');
  reset(fin);
  assign(fout, 'minim.out');
  rewrite(fout);
  
  if eof(fin) then
  begin
    close(fin);
    close(fout);
    exit;
  end;
  
  read(fin, n);
  for i := 1 to n do
    read(fin, v[i]);
    
  jumatate := n div 2;
  if jumatate = 0 then
    jumatate := 1;
    
  writeln(fout, afla_minim(jumatate, v));
  
  close(fin);
  close(fout);
end.
